core/src/main/java/nl/nn/adapterframework/receivers/JavaListener.java

```diff
@@ -144,45 +151,40 @@ public String processRequest(String correlationId, String rawMessage, HashMap co
 
 	@Override
 	public Message processRequest(Message message, @Nonnull PipeLineSession session) throws ListenerException {
-		@SuppressWarnings({"unchecked", "deprecation"})
-		RawMessageWrapper<M> rawMessageWrapper = new RawMessageWrapper<>((M)message.asObject(), session.getMessageId(), session.getCorrelationId());
+		MessageWrapper<M> rawMessageWrapper = new MessageWrapper<>(message, session.getMessageId(), session.getCorrelationId());
 		Message response = processRequest(rawMessageWrapper, message, session);
 		response.closeOnCloseOf(session, this);
 		return  response;
 	}
 
-	private Message processRequest(RawMessageWrapper<M> rawMessageWrapper, Message message, Map<String, Object> context) throws ListenerException {
+	private Message processRequest(@Nonnull RawMessageWrapper<M> rawMessageWrapper, @Nonnull Message message, @Nonnull PipeLineSession session) throws ListenerException {
 		if (!isOpen()) {
 			throw new ListenerException("JavaListener [" + getName() + "] is not opened");
 		}
-		log.debug("JavaListener [{}] processing correlationId [{}]" , getName(), context != null ? context.get(PipeLineSession.CORRELATION_ID_KEY) : null);
-		if (context != null) {
-			Object object = context.get("httpRequest");
-			if (object != null) {
-				if (object instanceof HttpServletRequest) {
-					ISecurityHandler securityHandler = new HttpSecurityHandler((HttpServletRequest)object);
-					context.put(PipeLineSession.SECURITY_HANDLER_KEY, securityHandler);
-				} else {
-					log.warn("No securityHandler added for httpRequest [{}]", object::getClass);
-				}
+		log.debug("JavaListener [{}] processing correlationId [{}]" , getName(), session.get(PipeLineSession.CORRELATION_ID_KEY));
+		Object object = session.get("httpRequest");
+		if (object != null) {
+			if (object instanceof HttpServletRequest) {
+				ISecurityHandler securityHandler = new HttpSecurityHandler((HttpServletRequest)object);
+				session.put(PipeLineSession.SECURITY_HANDLER_KEY, securityHandler);
+			} else {
+				log.warn("No securityHandler added for httpRequest [{}]", object::getClass);
 			}
 		}
-		try (PipeLineSession session = new PipeLineSession(context)) {
```

Comment:
We do not need to create a session here since the session we get is already good to use, from all callers.